interfaces/builtin/cups.go

```diff
@@ -39,7 +39,9 @@ const cupsBaseDeclarationSlots = `
     allow-installation:
       slot-snap-type:
         - app
-    deny-connection: true
+        - core
+    deny-connection:
+      on-classic: false
     deny-auto-connection: true
```

Comment:
I suggest (if I understand the code correctly) here to use
```
deny-auto-connection: false
```
as we are introducing the `cups` interface so that application Snaps with print functionality (like LibreOffice) plug to `cups` and this gets auto-connected as it does not allow admin tasks on CUPS. Only `cups-control`allows admin tasks and therefore should only auto-connect with permission from the Snap Store team.

Reply:
I left this out of the initial PR so the base declaration would match that of `cups-control`.  I agree that it would be good to have a conversation about loosening the restrictions for the `cups` slot, although I'm not sure if it should be in this PR.

At most, I think we'd want to a rule similar to the `x11` and `wayland` slots to allow auto connection on classic systems:

```
deny-auto-connection:
  on-classic: false
```

One issue to consider though is how many systems are there where `cups` gives meaningfully less access than `cups-control`?

Reply:
If CUPS is the CUPS Snap then there is always the difference, if CUPS is classically installed from Ubuntu's packages, the CUPS daemon does exactly the same checks, so if the client is a fully confined Snap and plugs only `cups` and not `cups-control` cupsd rejects administrative inquiries.

Reply:
What we want to have is that if someone uploads a Snap to the Snap Store and it plugs `cups`, the `cups` interface should auto-connect to both the system's and the CUPS Snap's `cups` slot on a user's system automatically, without that the uploader of the Snap needs to request permission from the Snap Store team.

Only if the uploaded Snap plugs `cups-control` the auto-connection (to both the system's and the CUPS Snap's `cups-control` slot) on the user's system should only happen if the uploader got permission from the Snap Store team.

This is a protection against malicious Snap Store uploads which mess with a user's printers and jobs whereas the common case of an application with which you can print is handled automatically without need of the intervention of the Snap Store team.

Reply:
From what I can tell, the cups-control restrictions currently only exist in Ubuntu >= 20.10 and the snap.  None of the LTS releases (where the majority of our users are) carry the patches.  I agree that we might want to loosen the restrictions in the future, but for now it looks `cups` carries the same risks as `cups-control` for the vast majority of users.

I think it is more important to have the slot available soon, and then tackle permissions later.

Reply:
I agree - even if we SRU this support back to the LTS releases, this still leaves all non-Ubuntu users at risk if their classic cups cannot perform the snap mediation - so for now I think it is best to keep `cups` with `deny-auto-connection: true` - OR the only other option I can think is that if snapd was somehow able to detect that the locally installed cups was able to perform the mediation *then* it would auto-connect cups in that case but I feel this is likely too complicated and therefore fraught so would prefer we stick to the current situation of having cups be manually connected.

Reply:
snapd could check whether the running CUPS is "secure" (having snap mediation) as there are tasks which CUPS considers administrative but do not change configuration, as `lpinfo -v` or `cupsctl`, so snapd could check the installed CUPS before deciding to auto-connect the `cups` interface.